Userland/Libraries/LibJS/Runtime/ArrayBuffer.h

```diff
@@ -213,11 +213,43 @@ void ArrayBuffer::set_value(size_t byte_index, Value value, [[maybe_unused]] boo
 {
     auto& vm = this->vm();
 
+    // 1. Assert: IsDetachedBuffer(arrayBuffer) is false.
+    VERIFY(!is_detached());
+
+    // 2. Assert: There are sufficient bytes in arrayBuffer starting at byteIndex to represent a value of type.
+    VERIFY(buffer().bytes().slice(byte_index).size() >= sizeof(T));
+
+    // 3. Assert: value is a BigInt if IsBigIntElementType(type) is true; otherwise, value is a Number.
+    if constexpr (IsIntegral<T> && sizeof(T) == 8)
+        VERIFY(value.is_bigint());
+    else
+        VERIFY(value.is_number());
+
+    // 4. Let block be arrayBuffer.[[ArrayBufferData]].
+    auto& block = buffer();
+
+    // FIXME: 5. Let elementSize be the Element Size value specified in Table 70 for Element Type type.
+
+    // 6. If isLittleEndian is not present, set isLittleEndian to the value of the [[LittleEndian]] field of the surrounding agent's Agent Record.
+    //    NOTE: Done by default parameter declaration of this function.
+
+    // 7. Let rawBytes be NumericToRawBytes(type, value, isLittleEndian).
     auto raw_bytes = numeric_to_raw_bytes<T>(vm, value, is_little_endian);
 
-    // FIXME: Check for shared buffer
+    // FIXME 8. If IsSharedArrayBuffer(arrayBuffer) is true, then
+    {
```

Comment:
`if (false) {`

Reply:
Ah sure! I wasn't sure what style LibJS would do for that, will use that :)